adobe/react-spectrum
# #3578 Add the ability to add a data attribute to <HiddenSelect /> to handle false a11y positives

Closes [#3578 ](https://github.com/adobe/react-spectrum/issues/3578)

This PR simply add the ability to expose a data-attribute on the `<HiddenElement />` of `useSelect`.

This is useful when using a tool such as AXE or pa11y giving you false positive - the data attribute is a good way to force the tool to ignore the false positive.

## ✅ Pull Request Checklist:


## 📝 Test Instructions:

1. Use the following code (from the `useSelect` documentation example), ensuring that the `props` contains `{hasA11yIgnoreDataAttr: true}`
2. Ensure that the hidden select element has the data attribute `data-rsp-a11y-ignore`.
3. Ensure unit tests are passing (`yarn jest`)

```tsx
import {HiddenSelect, useSelect} from '@react-aria/select';
import {Item} from '@react-stately/collections';
import {useSelectState} from '@react-stately/select';

// Reuse the ListBox, Popover, and Button from your component library. See below for details.
import {ListBox, Popover, Button} from 'your-component-library';

function Select(props) {
  // Create state based on the incoming props
  let state = useSelectState(props);

  // Get props for child elements from useSelect
  let ref = React.useRef(null);
  let {
    labelProps,
    triggerProps,
    valueProps,
    menuProps
  } = useSelect(props, state, ref);

  return (
    <div style={{display: 'inline-block'}}>
      <div {...labelProps}>{props.label}</div>
      <HiddenSelect
        isDisabled={props.isDisabled}
        state={state}
        triggerRef={ref}
        label={props.label}
        name={props.name} />
      <Button
        {...triggerProps}
        buttonRef={ref}
        style={{height: 30, fontSize: 14}}>
        <span {...valueProps}>
          {state.selectedItem
            ? state.selectedItem.rendered
            : 'Select an option'
          }
        </span>
        <span
          aria-hidden="true"
          style={{paddingLeft: 5}}>
          ▼
        </span>
      </Button>
      {state.isOpen &&
        <Popover state={state} triggerRef={ref} placement="bottom start">
          <ListBox
            {...menuProps}
            state={state} />
        </Popover>
      }
    </div>
  );
}

<Select label="Favorite Color">
  <Item>Red</Item>
  <Item>Orange</Item>
  <Item>Yellow</Item>
  <Item>Green</Item>
  <Item>Blue</Item>
  <Item>Purple</Item>
  <Item>Black</Item>
  <Item>White</Item>
  <Item>Lime</Item>
  <Item>Fushsia</Item>
</Select>
```

## 🧢 Your Project:

N/A
